### PR TITLE
Standardize code to F2008

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
I had to use F2003 because certain features were relying on it. Code can also be compiled on GCC 10 now.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
Tests pass with the new `-std` flag.

## Checklist
_Put an `x` in the boxes that apply._

- [ ] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: src/cgns_utilities.F90 [hunk 1]
```diff
@@ -2603,7 +2603,7 @@ subroutine convertPlot3d(pFile, cFile)
     integer(kind=4) :: nZones, i, j, k, cg, dims(9), zoneID, ierr, Cx, Cy, Cz, iZone
     integer(kind=4), dimension(:, :), allocatable :: sizes
     real(kind=8), dimension(:, :, :), allocatable :: coorX, coorY, coorZ
-    character*12 :: zoneName
+    character(len=12) :: zoneName
```
Comment (nwu63) [inline src/cgns_utilities.F90:2606]: Please check that this is correct

File: src/cgns_utilities.F90 [hunk 2]
```diff
@@ -2622,7 +2622,7 @@ subroutine convertPlot3d(pFile, cFile)
     zoneLoop: do iZone=1, nZones
 
        ! Write the zone itself.
-       write(zoneName, "((a) (I5))")  'Domain.', izone
+       write(zoneName, "((a), (I5))")  'Domain.', izone
```
Comment (nwu63) [inline src/cgns_utilities.F90:2625]: Also this

File: src/cgns_utilities.F90 (hunk 1 above)
Comment (akleb) [reply src/cgns_utilities.F90:2606]: I came to the same independent conclusion for updating the code.

**Comment by akleb** [comment]
If we are planning on eventually updating to 2008 on all our codes, then would it makes sense to just knock that out here? I suspect it may already be 2008 compliant, or at least very close.

File: src/cgns_utilities.F90 (hunk 2 above)
Comment (akleb) [reply src/cgns_utilities.F90:2625]: The comma makes sense to me, but are the parentheses required? I am not sure what they are doing.

Comment (nwu63) [reply src/cgns_utilities.F90:2625]: I am not familiar with file IO specifications, not really sure :/ maybe @anily knows?

**Review by marcomangano — APPROVED** [review]
If the tests are working and the update is complete, I am fine with merging this. I am just changing the PR title to avoid confusion

**Review by akleb — APPROVED** [review]
I think this is all fine, looks the same as what I had for my corrections to pyHyp. As for the parentheses question I think they just [signify groupings](https://pages.mtu.edu/~shene/COURSES/cs201/NOTES/chap05/format.html), which in this case doesn't matter so it should be fine with or without them.

**Pull request merged**